sdk/core/Azure.Core.TestFramework/src/Azure.Core.TestFramework.csproj

```diff
@@ -11,6 +11,8 @@
     <PackageReference Include="Microsoft.NET.Test.Sdk" />
     <PackageReference Include="Moq" />
     <Reference Include="System.Web" />
+    <ProjectReference Include="../../../resources/Azure.ResourceManager.Resources/src/Azure.ResourceManager.Resources.csproj" />
```

Comment:
We generally try to avoid project references for things across different sdk directories as we generally scope builds to a given sdk service directory. Is there a published version of this that can be referenced via PackageReference? If not we should be sure to get one published and referenced as a package ASAP.

Reply:
I attempted this in a previous commit. https://github.com/Azure/azure-sdk-for-net/pull/15434/commits/c4a798eee3d64b475aa23196370d3c6202a621fe

This broke the test for resource group operations since it was now using the pulled in nuget package which is different from the current unreleased code in resource group operations.

Reply:
It means we would never be able to change the dependency to be a package reference because you won't be able to use the test framework to test `Azure.ResourceManager.Resources` itself. Should we hand write a small resource client only for operations that we need in the framework?

Reply:
Is this only needed for resource group creation? Or do we use for other creation as well? 

If it is only for resource group creation then we could create a small helper that would do the creation and deletion to help avoid this dependency. 

We could also consider creating an extension point that would allow folks to pass in a a specific resource group creator in order to abstract this dependency.

Reply:
One other potential option would to have an msbuild property that we could use to toggle this between a package or project reference depending on whether or not we needed the live version of this reference or if the package reference would be enough.

Reply:
@weshaggard  I think your last suggestion there, "One other potential option would to have an msbuild property that we could use to toggle..." would be the most straightforward.